### ISSUE TITLE
postprocessTree addon hook receives minified CSS files (when it is enabled)

### ISSUE THREAD

**Issue by zackthehuman** [ember-cli/ember-cli#6285]
The `postprocessTree` addon hook will receive a node of minified CSS files when minification is enabled (production builds, etc.). I find this confusing and a bit trolling, since it isn't documented what one should expect to receive in `postprocessTree`.

At first glace I assumed that I would get a node with files that were processed by the preprocessors and possibly moved (e.g. `'app/styles/app.css' -> 'assets/example.css'`). I assumed that minification happened at the *very end*. I spoke with another developer today who also had the same assumption and made mistakes similar to mine.

**Why I find this confusing:**

I have an addon that uses `rtlcss` via `postcss`. `rtlcss` can be controlled in a CSS file via special comments, similar to how you might disable specific linting rules when you know you need to. Since `postcss` is something you generally do near the end of your pipeline it seemed to make sense to put that processing in `postprocessTree`. However, since the node is already minified by the time the hook is run, the special comments for `rtlcss` have been stripped, resulting in incorrect processing.

Basically, everything works fine if minification is turned off, which results in a bug that only surfaces in production builds or builds where minification is specifically enabled.

**My workaround:**

I am currently using `setupPreprocessorRegistry` coupled with some addon initialization order settings to make my addon work. It is not ideal, and is definitely confusing for other developers who look at the code.

**Possible suggestions:**

I think we need another hook. A hook that runs after preprocessing and moving/merging, but before minification. This way it would be clear where in the pipeline you are if you need to do something like `postcss` before you minifiy.

**Comment by nathanhammond** [comment]
Agreed, this needs work and specification. [Here is that tracking issue.](https://github.com/ember-cli/rfcs/issues/64) And now this issue is linked to from the tracking issue.

**Comment by stefanpenner** [comment]
> I assumed that minification happened at the very end.

What defines the "very end" is a nebulous concept and fraught with peril, remember asset finger printing happens after uglification which means it also does not run at "the very end". 

this and https://github.com/ember-cli/ember-cli/issues/6286 seem very related. 

https://github.com/ember-cli/ember-cli/issues/6286#issuecomment-248665859 is the same comment I would make here.

i'm going to close this, because I do not believe these two issues are unrelated.
